pyccel/codegen/printing/cwrappercode.py

```diff
@@ -1172,7 +1178,9 @@ def _print_Module(self, expr):
         interface_funcs = [f.name for i in expr.interfaces for f in i.functions]
         funcs = [*expr.interfaces, *(f for f in funcs_to_wrap if f.name not in interface_funcs)]
 
+        self._in_header = True
         decs = ''.join('extern '+self._print(d) for d in vars_to_wrap_decs)
+        self._in_header = False
```

Comment:
Is this a bug fix?

Reply:
Yes and no, these variables need to be declared as though they were in a header. But in reality the only place where that changes something is when printing arrays. In this case we need `self._in_header` to print:
`ndarray_t a;` instead of `ndarray_t a = {.shape = NULL};`